File: sameTree.cpp
//
//  sameTree.cpp
//  leetcode
//
//  Created by Kael on 15/8/31.
//  Copyright (c) 2015年 Kael. All rights reserved.
//

#include <stdio.h>

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    bool isSameTree(TreeNode p, TreeNode q) {
        return true;
    }
};

File: go/same-tree.go
package main

import (
	"fmt"
)

func main() {
	a1 := &TreeNode{
		Val: 0,
	}
	b1 := &TreeNode{
		Val: 0,
	}
	fmt.Println(isSameTree(a1, b1))
}

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isSameTree(p *TreeNode, q *TreeNode) bool {
	if p == nil || q == nil {
		if p == q {
			return true
		}
		return false
	}
	return p.Val == q.Val && isSameTree(p.Left, q.Left) && isSameTree(p.Right, q.Right)
}
